code-challenges.js

```diff
@@ -12,64 +13,140 @@ var num2 = 0
 
 // --------------------3) Create a function that takes in the mixedDataArray array and returns an array with ONLY NUMBERS sorted from least to greatest.
+console.log("3:");
 
 var mixedDataArray = [true, 8, "hello", 90, -8, null, 0, 46, 59, 107, "hey!"]
 // Expected output: [-8, 0, 8, 46, 59, 90, 107]
 
+const onlyNumbersSort = (array) => {
+    // filter out non-number types
+    let filteredArray = array.filter(x => typeof x === "number");
+    // sort new array of only numbers
+    let sortedArray = filteredArray.sort((a, b) => a - b);
+    return sortedArray
+}
 
-
-
+console.log(onlyNumbersSort(mixedDataArray), "\n");
 
 // --------------------4) Create a function that takes in a string and logs the index of the first vowel.
+console.log("4:");
 
 var vowelTester1 = "learn"
 // Expected output: 1
 var vowelTester2 = "throw"
 // Expected output: 3
 
-
-
-
+const firstVowelIndex = (string) => {
+    let word = string.toLowerCase();
+    let vowelArr = [];
+    // deconstruct string into array of letters
+    let  arrLetters = word.split("");
+    // map through array checking for vowels, if found, push index to vowelArr
+    arrLetters.map((letter, index) => {
+        if(letter === "a" || letter === "e" || letter === "i" || letter === "o" || letter === "u" || letter === "y") {
+            vowelArr.push(index);
+        }
+    })
+    // return first value in vowelIndex
+    return vowelArr[0];
+}
+
+console.log(firstVowelIndex(vowelTester1));
+console.log(firstVowelIndex(vowelTester2), "\n");
 
 
 // --------------------5) Create a function that takes three arguments - two numbers and a mathematical operation (+, -, *, /) and performs the given calculation. If the input tries to divide by 0, return: "Can't divide by 0!"
-
-
-
-
+console.log("5:");
+
+const calculator = (num1, operator, num2) => {
+    // Type check for numbers
+    if(typeof num1 === "number" && typeof num2 === "number"){
+    // If else statements for each operator 
+        if(operator === "+"){
+            return num1 + num2;
+        } else if(operator === "-"){
+            return num1 - num2;
+        } else if(operator === "*"){
+            return num1 * num2;
+        } else if(operator === "/"){
+            // if statement accounting for 0's
+            if(num2 === 0){
+                return "Can't divide by 0!";
+            } else {
+                return num1 / num2;
+            }
+        } else {
+            return "Please input two numbers and a mathematical operator"
+        }
+    } else {
+        return "Please input two numbers and a mathematical operator"
+    }
+}
 
 // Uncomment and use the following console logs to test your function
-// console.log(calculator(3, "*", 9))
+console.log(calculator(3, "*", 9))
 // Expected output: 27
 
-// console.log(calculator(16, "+", 3))
+console.log(calculator(16, "+", 3))
 // Expected output: 19
 
-// console.log(calculator(89, "/", 0))
+console.log(calculator(89, "/", 0), "\n")
```

Comment:
* *chefs kiss* * WONDERFUL. Love that you check for typeof ahead of time before going into your logic, helps keep it DRY! :)